### PR TITLE
[Hub Generated] Review request for Microsoft.Confluent to add version preview/2020-03-01-preview

### PR DESCRIPTION
**Pull request by alluri02**
This is a PR generated at [OpenAPI Hub](https://aka.ms/openapihub). You can view your [work branch via this link](https://portal.azure-devex-tools.com/branch/alluri02/azure-rest-api-specs/confluent-terms-api...master/).

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of “no breaking changes
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [x] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “**WaitForARMFeedback**” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [x] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from API Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable version with Breaking Change Validation errors
- [ ] Updating API(s) in preview over 1 year

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/confluent/resource-manager/Microsoft.Confluent/preview/2020-03-01-preview/confluent.json [hunk 1]
```diff
@@ -33,6 +33,78 @@
...
+              "nextLinkName": "nextLink"
+          },
+          "x-ms-examples": {
+            "MarketplaceAgreements_List": {
+              "$ref": "./examples/MarketplaceAgreements_List.json"
+            }
+          }
+        }
+     },
+     "/subscriptions/{subscriptionId}/providers/Microsoft.Confluent/agreements/all/acceptTerms": {
```
Comment (majastrz) [inline specification/confluent/resource-manager/Microsoft.Confluent/preview/2020-03-01-preview/confluent.json:75]: This API is problematic from ARM's perspective. See the following comment on a different PR: https://github.com/Azure/azure-rest-api-specs-pr/pull/2309#discussion_r514506693. I'm starting a mail thread to discuss this...

File: specification/confluent/resource-manager/Microsoft.Confluent/preview/2020-03-01-preview/confluent.json [hunk 2]
```diff
@@ -33,6 +33,88 @@
...
+          },
+          "x-ms-examples": {
+            "MarketplaceAgreements_List": {
+              "$ref": "./examples/MarketplaceAgreements_List.json"
+            }
+          }
+        }
+     },
+     "/subscriptions/{subscriptionId}/providers/Microsoft.Confluent/agreements/default": {
+          "post": {
```
Comment (chiragg4u) [inline specification/confluent/resource-manager/Microsoft.Confluent/preview/2020-03-01-preview/confluent.json:76]: I thought we agreed to use PUT for these?

**Comment by chiragg4u** [comment]
Please fix the above comment.
Work with SDK team for the breaking change, process given @ https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/202/Overall-Process-of-AME-Onboarding

**Comment by ruowan** [comment]
Not release yet. No customer impact

**Review by ruowan — APPROVED** [review]

**Pull request merged**